### PR TITLE
Initial commit for MWM++ architecture

### PR DESCRIPTION
**Pull request by lucasponce**
- Basic layout using HawkularServer (Vert.X based server)
- Add Commons and Inventory projects

### CODE REVIEW

**Comment by lucasponce** [comment]
@jshaughn @jotak @jmazzitelli

**Comment by jotak** [comment]
@lucasponce I didn't realize that you were adding pieces of commons here. Since I'm modifying commons in the same time it's out of sync.
So the idea is to get everything about mwm poc in a single repo?

**Comment by lucasponce** [comment]
@jotak well, yes, the HawkularServer is and other common utils are hosted in commons and in this initial attempt I used them just to replicate the same stack we added in alerting.
I see the issue, so, perhaps we might put everything under the single repo for poc and we can move code later.
I think that in the current discussions perhaps the inventory model API will be only used from the hServices, so yes, it could make sense.

**Comment by lucasponce** [comment]
@jshaughn could you review this ? @jotak would need it  to consolidate things

File: hawkular-rest/hawkular-listener/src/main/java/org/hawkular/listener/MIQEventUtils.java [hunk 1]
```diff
@@ -1,77 +0,0 @@
...
-import java.util.HashSet;
-import java.util.Set;
-
-import org.hawkular.inventory.paths.CanonicalPath;
-import org.hawkular.listener.bus.ListenerUtils;
-
-/**
- * Group together MIQ events sent to alerting
- */
-public class MIQEventUtils {
```
Comment (jshaughn) [inline hawkular-rest/hawkular-listener/src/main/java/org/hawkular/listener/MIQEventUtils.java]: TODO : What happens in this class will still be important going forward.

File: hawkular-rest/hawkular-listener/src/main/java/org/hawkular/listener/bus/CommandEventListener.java [hunk 2]
```diff
@@ -1,90 +0,0 @@
...
- * payload should be a command pojo.  We want to generate only one Hawkular Event per Command Gateway event, so this
- * is Queue based, limiting message consumption to one server.
- * </p>
- * @author Jay Shaughnessy
- */
-@MessageDriven(messageListenerInterface = MessageListener.class, activationConfig = {
-        @ActivationConfigProperty(propertyName = "destinationType", propertyValue = "javax.jms.Queue"),
-        @ActivationConfigProperty(propertyName = "destination", propertyValue = "HawkularCommandEvent") })
-@TransactionAttribute(value = TransactionAttributeType.NOT_SUPPORTED)
-public class CommandEventListener extends BasicMessageListener<BasicMessage> {
```
Comment (jshaughn) [inline hawkular-rest/hawkular-listener/src/main/java/org/hawkular/listener/bus/CommandEventListener.java]: TODO : What happens in this class will still be important going forward.

File: hawkular-rest/hawkular-listener/src/main/java/org/hawkular/listener/bus/ListenerUtils.java [hunk 3]
```diff
@@ -1,103 +0,0 @@
...
-import java.util.UUID;
-
-import javax.naming.InitialContext;
-
-import org.hawkular.alerts.api.model.event.Event;
-import org.hawkular.alerts.api.services.AlertsService;
-import org.hawkular.inventory.paths.CanonicalPath;
-import org.jboss.logging.Logger;
-
-public class ListenerUtils {
```
Comment (jshaughn) [inline hawkular-rest/hawkular-listener/src/main/java/org/hawkular/listener/bus/ListenerUtils.java]: TODO : What happens in this class will still be important going forward.

**Pull request merged**